asciidoc/courses/llm-fundamentals/modules/2-vectors-semantic-search/lessons/1-semantic-search/lesson.adoc

```diff
@@ -0,0 +1,68 @@
+= Vectors & Semantic Search
+
+// TODO: Rewrite
+In the last lesson, we discussed the importance of grounding to improve LLM accuracy and the concept of **Retrieval Augmented Generation** (RAG).
+RAG involves providing additional information along with a prompt to help the LLM form a response.
+
+One of the big challenges of RAG is understanding what the user is asking for, and to surface the right information to pass to the LLM.
+
+
+== Semantic Search vs. Traditional Search
+
+At its core, semantic search aims to understand the intent and contextual meaning of search phrases, rather than just focusing on individual keywords.
+Traditional search often depends on exact match keywords or proximity-based algorithms to return results.
+
+For example, if you input "apple" in a traditional search, you might predominantly get results about the fruit.
+However, in a semantic search, the engine tries to gauge the context: Are you searching about the fruit, the tech company, or something else?
+
+The results are hence tailored not just based on the term, but the perceived intent behind it.
+
+== Vectors and embeddings
+
+
+In the realm of natural language processing (NLP) and machine learning, words or phrases are often converted into numerical representations known as **vectors**.
+Each dimension in a vector can represent a particular semantic aspect of the word or phrase.
+With advanced models, these embeddings also contain contextual information.
+This means the vector for a word can change based on its surrounding context. For instance, the word _bank_ will have a different vector in _river bank_ than in _savings bank_.
+Semantic search systems can use these contextual embeddings to better understand user intent.
+
+
+[INFO]
+.Creating Vector Embeddings
+====
+All LLM providers expose API endpoints that covert a _chunk_ of text into a vector embedding.
+Depending on the provider, the shape of the vector may differ.
+
+OpenAI's `text-embedding-ada-002` embedding model converts text into a vector consisting of 1,536 dimensions.
+====
+
+The _distance_ or _angle_ between vectors can be used to gauge the semantic similarity between words or phrases.
```

Comment:
Add a sub section "=== Similarity Vector Search" here

Reply:
This section is hard to follow. There are a lot of new concepts. A diagram here would be really helpful to show the difference between the concepts and how they relate.